### PR TITLE
Add support for uniprot reference database

### PR DESCRIPTION
**Pull request by rababerladuseladim**
This pull requests adds support for using [uniprot](https://www.uniprot.org/) as a reference database.

# Changes

### dudesdb
- reference mode `up` for processing uniprot fasta files

### dudes
- input format `custom_blast_file`, requiring a tabular file with the columns 'qseqid', 'sseqid', 'slen', 'sstart', 'evalue' as created by the [diamond sequence aligner](https://github.com/bbuchfink/diamond)
- CLI option `--debug` for printing debug info to STDERR
- CLI option `--debug_plots_dir`: create plots for each score comparison in this directory

## changed
- migrated from setup.py to poetry based installation
- continuous integration tests
- renamed script entrypoint DUDes.py to dudes
- renamed script entrypoint DUDesDB.py to dudesdb

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,150 +41,148 @@ Usage:
...
+dudesdb -m 'av' -f references.fasta -n nodes.dmp -a names.dmp -g nucl_gb.accession2taxid -t 12 -o custom_db
+```
 
-	New NCBI header [>NC_009925.1 Acaryochloris marina MBIC11017, complete genome.]
-		-m 'av'
-	Old NCBI header [>gi|158333233|ref|NC_009925.1| Acaryochloris marina MBIC11017, complete genome.]
-		-m 'gi'
+- Choose the parameter `-m` considering the format of the headers in your reference sequences:
+	- `-m 'av'`: New NCBI header [>NC_009925.1 Acaryochloris marina MBIC11017, complete genome.]
+	- `-m 'gi'`: Old NCBI header [>gi|158333233|ref|NC_009925.1| Acaryochloris marina MBIC11017, complete genome.]
```
Comment (pirovc) [inline README.md:121]: Add here an example for the `up` mode

File: README.md [hunk 2]
```diff
@@ -1,41 +1,37 @@
...
+## Toy example with sample data:
 
-- Download the pre-compiled index and database:
+```sh
+dudes -s sampledata/hiseq_accuracy_k60.sam -d sampledata/arc-bac_refseq-cg_201503.npz -o sampledata/dudes_profile_output
+```
+
+- The sample data is based on a set of bacterial whole-genome shotgun reads comprising 10 organisms (HiSeq - 10000 reads [1]). The read set was mapped with Bowtie2 [2] against the set of complete genome sequences (dudesdb_arc-bac_refseq-cg_201503).
+
+## Example with pre-compiled DB for metagenomics:
```
Comment (pirovc) [inline README.md:31]: Add maybe a similar example for metaproteomics data

File: README.md (hunk 1 above)
Comment (rababerladuseladim) [reply README.md:121]: Added in 3886805a142f15ecf5c3c0a285ba6b0584311db9

File: README.md (hunk 2 above)
Comment (rababerladuseladim) [reply README.md:31]: I'll try but the zipped DB for uniprot ist 54 GB, with the size limit of zenodo at 50 GB per dataset :see_no_evil:

Comment (pirovc) [reply README.md:31]: Sorry, I wasn't clear enough. I meant only an example on how to run it with metaproteomic data, not necessarily a pre-compiled database.

**Pull request merged**